java-type-checker/java_type_checker/types.py

```diff
@@ -89,6 +89,9 @@ class JavaPrimitiveType(JavaType):
     Primitive types are not object types and do not have methods.
     """
 
+    def is_subtype_of(self, other):
+        return (self.name == other.name) and (other.__class__ == JavaPrimitiveType)
```

Comment:
Your version of this method is safer because it doesn't rely on the names of types, which could be the same